native/python/pyjp_monitor.cpp

```diff
@@ -35,27 +35,37 @@ static int PyJPMonitor_init(PyJPMonitor *self, PyObject *args)
 	JP_PY_TRY("PyJPMonitor_init");
 	self->m_Monitor = NULL;
 	JPContext *context = PyJPModule_getContext();
-	JPJavaFrame frame(context);
+	JPJavaFrame frame = JPJavaFrame::outer(context);
 
 	PyObject* value;
 
 	if (!PyArg_ParseTuple(args, "O", &value))
-	{
 		return -1;
-	}
 
 	JPValue *v1 = PyJPValue_getJavaSlot(value);
 	if (v1 == NULL)
-		JP_RAISE(PyExc_TypeError, "Java object is required.");
+	{
+		PyErr_SetString(PyExc_TypeError, "Java object is required.");
+		return -1;
+	}
 
 	if (v1->getClass() == context->_java_lang_String)
-		JP_RAISE(PyExc_TypeError, "Java strings cannot be used to synchronize.");
+	{
+		PyErr_SetString(PyExc_TypeError, "Java strings cannot be used to synchronize.");
+		return -1;
+	}
 
 	if ((v1->getClass())->isPrimitive())
-		JP_RAISE(PyExc_TypeError, "Java primitives cannot be used to synchronize.");
+	{
+		PyErr_SetString(PyExc_TypeError, "Java primitives cannot be used to synchronize.");
+		return -1;
+	}
```

Comment:
There were lots of spots where the C++ exception path (slow) was being taken just to dump a Python exception back.   This is unnecessary so all these were removed.